internal/sim/components.go

```diff
@@ -22,24 +22,121 @@ import (
 
 //go:generate ../../cmd/weaver/weaver generate .
 
-// This file contains components used to test the simulator in sim_test.go.
+// This file contains components used to test the simulator in sim_test.go. The
+// callgraph is intentionally overcomplicated to exercise the simulator.
 
-type swapper interface {
-	Swap(context.Context, int, int) (int, int, error)
+// Component interfaces.
+
+type divMod interface {
+	// DivMod(n, d) returns n/d, n%d.
```

Comment:
I thought that you are computing `(n/d) % d`, which is an awesomer (and more useless) operation. :smile:

Although, maybe I just came up with a tool to solve the collatz conjecture! :rofl:

Reply:
:laughing: I took the idea from Python's [divmod](https://docs.python.org/3/library/functions.html#divmod).